stdlib/2and3/warnings.pyi

```diff
@@ -1,33 +1,37 @@
 # Stubs for warnings
 
-# Based on http://docs.python.org/3.2/library/warnings.html
+from typing import Any, Dict, NamedTuple, Optional, TextIO, Union
+from types import ModuleType, TracebackType
 
-from typing import Any, List, TextIO, Union
-
-def warn(message: Union[str, Warning], category: type = ...,
+def warn(message: Union[str, Warning], category: Optional[type] = ...,
          stacklevel: int = ...) -> None: ...
-
 def warn_explicit(message: Union[str, Warning], category: type, filename: str,
-                  lineno: int, module: str = ..., registry: Any = ...,
-                  module_globals: Any = ...) -> None: ...
-
-# logging modifies showwarning => make it a variable.
-def _showwarning(message: str, category: type, filename: str, lineno: int,
-                 file: TextIO = ..., line: str = ...) -> None: ...
-showwarning = _showwarning
-
+                  lineno: int, module: Optional[str] = ...,
+                  registry: Optional[Dict[Any, Any]] = ...,
```

Comment:
Yeay, it's not clearly documented, I don't think that `warn_explicit` should even be exposed or at least, it should be more documented.